File: Advanced/duediligence-cordapp/contracts/src/main/java/net/corda/samples/duediligence/states/CorporateRecordsAuditRequest.java
package net.corda.samples.duediligence.states;

import net.corda.core.contracts.LinearState;
import net.corda.core.contracts.UniqueIdentifier;
import net.corda.core.serialization.ConstructorForDeserialization;
import net.corda.samples.duediligence.contracts.CorporateRecordsContract;
import net.corda.core.contracts.BelongsToContract;
import net.corda.core.contracts.ContractState;
import net.corda.core.identity.AbstractParty;
import net.corda.core.identity.Party;
import org.jetbrains.annotations.NotNull;

import java.util.Arrays;
import java.util.List;

// *********
// * State *
// *********
@BelongsToContract(CorporateRecordsContract.class)
public class CorporateRecordsAuditRequest implements LinearState {
    /*This can include reviewing incorporation documents, company constitutions,
    organisational charts, a list of security holders, employee share plans and any
    options granted to acquire securities.*/

    //private variables
    private Boolean qualification = false;
    private Party applicant;
    private Party validater;
    private int numberOfFiles;
    private UniqueIdentifier linearId;

    /* Constructor of your Corda state */
    @ConstructorForDeserialization
    public CorporateRecordsAuditRequest(Party applicant, Party validater, int numberOfFiles, UniqueIdentifier linearId, Boolean qualification) {
        this.applicant = applicant;
        this.validater = validater;
        this.numberOfFiles = numberOfFiles;
        this.linearId = linearId;
        this.qualification = qualification;
    }

    public CorporateRecordsAuditRequest(Party applicant, Party validater, int numberOfFiles, UniqueIdentifier linearId) {
        this.applicant = applicant;
        this.validater = validater;
        this.numberOfFiles = numberOfFiles;
        this.linearId = linearId;
    }

    public CorporateRecordsAuditRequest(Party applicant, Party validater, int numberOfFiles) {
        this.qualification = qualification;
        this.applicant = applicant;
        this.validater = validater;
        this.numberOfFiles = numberOfFiles;
        this.linearId = new UniqueIdentifier();
    }

    //getters
    public Boolean getQualification() {        return qualification;    }
    public Party getApplicant() {        return applicant;    }
    public Party getValidater() {        return validater;    }
    public int getNumberOfFiles() {        return numberOfFiles;    }

    /* This method will indicate who are the participants and required signers when
     * this state is used in a transaction. */
    @Override
    public List<AbstractParty> getParticipants() {
        return Arrays.asList(applicant,validater);
    }

    public void validatedAndApproved (){
        this.qualification = true;
    }

    @NotNull
    @Override
    public UniqueIdentifier getLinearId() {
        return this.linearId;
    }
}

File: kotlin-samples/Advanced/duediligence-cordapp/contracts/src/main/kotlin/net/corda/samples/duediligence/states/CorporateRecordsAuditRequest.kt
package net.corda.samples.duediligence.states

import net.corda.core.contracts.BelongsToContract
import net.corda.core.contracts.LinearState
import net.corda.core.contracts.UniqueIdentifier
import net.corda.core.identity.AbstractParty
import net.corda.core.identity.Party
import net.corda.samples.duediligence.contracts.CorporateRecordsContract

/*This can include reviewing incorporation documents, company constitutions,
organisational charts, a list of security holders, employee share plans and any
options granted to acquire securities.*/
@BelongsToContract(CorporateRecordsContract::class)
class CorporateRecordsAuditRequest(
        var qualification: Boolean = false,
        val applicant: Party,
        val validater: Party,
        val numberOfFiles: Int = 0,
        override val participants: List<AbstractParty> = listOf(applicant,validater),
        override val linearId: UniqueIdentifier = UniqueIdentifier()
) : LinearState {

    fun validatedAndApproved() {
        qualification = true
    }
}